doc/document.go

```diff
@@ -138,7 +138,7 @@ func (d Document) Equal(other Document) bool {
 
 // Validate returns a bool indicating whether the document is valid.
 func (d Document) Validate() error {
-	if len(d.Fields) == 0 {
+	if len(d.Fields) == 0 && !d.HasID() {
```

Comment:
maybe just a test for this case if its easy? I have to-do to go back and improve test coverage soon so can also get it then

Reply:
added